app/src/main/java/com/example/henzoshimada/feeltrip/ElasticSearchController.java

```diff
@@ -32,18 +33,30 @@
     private static final String typeMood = "mood";
     private static final String typeUser = "user";
 
-    public static class AddMoodTask extends AsyncTask<Mood, Void, Void> {
+    public static class AddMoodTask extends AsyncTask<Mood, Void, Boolean> {
+
+        private Context context;
+
+        public AddMoodTask(Context context) {
+            this.context = context.getApplicationContext();
+        }
 
         @Override
-        protected Void doInBackground(Mood ... moods ) {
+        protected void onPostExecute(Boolean result) {
+            FeelTripApplication.loadFromElasticSearch();
+            FeelTripApplication.getMoodAdapter(context).notifyDataSetChanged();
+        }
+
+        @Override
+        protected Boolean doInBackground(Mood ... moods ) {
             if(android.os.Debug.isDebuggerConnected())
                 android.os.Debug.waitForDebugger();
             verifySettings();
 
             for (Mood mood : moods) {
                 mood.setMade(new Date()); // document around the instant the mood is actually built and posted onto elasticsearch
                 if(mood.getMade() != null) {
-                    Index index = new Index.Builder(mood).index(groupIndex).type(typeMood).build();
+                    Index index = new Index.Builder(mood).index(groupIndex).type(typeMood).refresh(true).build();
```

Comment:
Jest has a "wait until elasticsearch finishes reindexing" command built right into it. Who knew?